core/Segment.php

```diff
@@ -110,6 +110,35 @@ public function __construct($segmentCondition, $idSites)
         }
     }
 
+    private function getAvailableSegments()
+    {
+        // segment metadata
+        if (empty($this->availableSegments)) {
+            $this->availableSegments = API::getInstance()->getSegmentsMetadata($this->idSites, $_hideImplementationData = false);
```

Comment:
I don't think it's worth it. When this class is used, it means we're fetching a report or something and then through the whole request we only work on one idSite. Or to describe it differently during one request it should not be called with different `idSites`. Also do not really see the advantage of putting it in DI apart from a possible caching. I'd rather add caching when there is an actual performance problem or so